spec/views/new_document/choose_document_type.html.erb_spec.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+RSpec.describe "new_document/choose_document_type.html.erb" do
+  describe "choose document type form" do
+    it "should render correctly for every document type" do
+      DocumentType.all.each do |document_type|
+        assign(:supertype, document_type.parent)
+        render
+        expect(rendered).to have_content(document_type.label)
+      end
```

Comment:
Loops are normally frowned upon tests, though I'm not sure there'll particularly nice way to do this without a loop as feature tests would need a lot of steps.


If we do need a loop it's probably better to do this with the it inside the loop. For example:
```
    DocumentType.all.each do |document_type|
      it "should render correctly for #{document_type.name}" do
        assign(:supertype, document_type.parent)
        render
        expect(rendered).to have_content(document_type.label)
      end
    end
```

Assuming there's not some loading related issue that stops you using DocumentType.all until runtime.

I assume this also is going to change very soon though?

Reply:
+1 for moving the `it` block inside the loop. I'll do that.

This will change, but not by very much - we'd probably iterate over a `DocumentTypeSelection` instead of a `DocumentType` and `assign(:document_type_selection, document_type_selection)`.